File: 品味Java的21種設計模式/dp_cpp/strategy/example5/Card2.cpp
#include "Card2.h"

namespace cn
{
	namespace javass
	{
		namespace dp
		{
			namespace strategy
			{
				namespace example5
				{

					Card2::Card2(std::string account)
					{
						InitializeInstanceFields();
						this->account = account;
					}

					void Card2::pay(PaymentContext *ctx)
					{
						puts("现在给"+ctx->getUserName()+"的"+this->account+"帐号支付了"+ctx->getMoney()+"元");
						//连接银行，进行转帐，就不去管了
					}
				}
			}
		}
	}
}

File: 品味Java的21種設計模式/dp_java/strategy/example5/Card2.java
package cn.javass.dp.strategy.example5;
/**
 * 支付到银行卡
 */
public class Card2 implements PaymentStrategy {
    /**
     * 帐号信息
     */
    private String account = "";
    /**
     * 构造方法，传入帐号信息
     * @param account 帐号信息
     */
    public Card2(String account) {
        this.account = account;
    }
    public void pay(PaymentContext ctx) {
        System.out.println("现在给" + ctx.getUserName() + "的" + this.account + "帐号支付了" + ctx.getMoney() + "元");
        //连接银行，进行转帐，就不去管了
    }
}
